### ISSUE TITLE
[Solved] Latest version Random crash on ubuntu 20.04 PPA

### ISSUE THREAD

**Issue by bakamosokyo** [c0re100/qBittorrent-Enhanced-Edition#186]
**Installed from PPA, nox ver.
Two VMs from different  providers experiencing random crash using latest version.
No problem before updating.** 

Qt: | 5.12.8
-- | --
Libtorrent: | 1.2.11.0
Boost: | 1.71.0
OpenSSL: | 1.1.1f
zlib: | 1.2.11

Some logs from systemctl status

_qbittorrent-nox[629]:   /usr/lib/x86_64-linux-gnu/libtorrent-rasterbar.so.10 : ()+0x332df1  [0x7fb1b6451df1]
qbittorrent-nox[629]:   /usr/lib/x86_64-linux-gnu/libtorrent-rasterbar.so.10 : ()+0x36b7ed  [0x7fb1b648a7ed]
systemd[1]: qbittorrent.service: Main process exited, code=killed, status=6/ABRT
qbittorrent-nox[629]:   /usr/lib/x86_64-linux-gnu/libtorrent-rasterbar.so.10 : boost::asio::detail::completion_handler<libtorrent::torrent_handle::async_call<void (libtorrent::torrent::*)(libtorrent::fla>
qbittorrent-nox[629]:   /usr/lib/x86_64-linux-gnu/libtorrent-rasterbar.so.10 : ()+0xe0d7d  [0x7fb1b61ffd7d]
qbittorrent-nox[629]:   /usr/lib/x86_64-linux-gnu/libtorrent-rasterbar.so.10 : ()+0x25d876  [0x7fb1b637c876]
qbittorrent-nox[629]:   /usr/lib/x86_64-linux-gnu/libstdc++.so.6 : ()+0xd6d84  [0x7fb1b5855d84]
qbittorrent-nox[629]:   /lib/x86_64-linux-gnu/libpthread.so.0 : ()+0x9609  [0x7fb1b6105609]
qbittorrent-nox[629]:   /lib/x86_64-linux-gnu/libc.so.6 : clone()+0x43  [0x7fb1b5545293]
systemd[1]: qbittorrent.service: Failed with result 'signal'._

**logs from ~/.local/share/data/qBittorrent/ shows nothing related at all.**

**No idea how to reproduce, But on both VMs simply pause torrent would kill it.
Some times it just crash Itself**

**All settings changed from default should not result a crash. Its all usual changes everybody do.**

**I dont know how can I see more logs, maybe somebody could tell me?
I will try to keep updating logs when crash happens.**

**Comment by c0re100** [comment]
ping @poplite

Maybe crashes related to boost library?
`qbittorrent-nox[629]: /usr/lib/x86_64-linux-gnu/libtorrent-rasterbar.so.10 : boost::asio::detail::completion_handler<libtorrent::torrent_handle::async_call<void (libtorrent::torrent::*)(libtorrent::fla>`

**Comment by poplite** [comment]
```
assertion failed. Please file a bugreport at https://github.com/arvidn/libtorrent/issues
Please include the following information:

version: 1.2.11.0-6597bd0f1

file: '../../src/torrent.cpp'
line: 7932
function: check_invariant
expression: current_stats_state() == int(m_current_gauge_state + counters::num_checking_torrents) || m_current_gauge_state == no_gauge_state
```

It seems I forgot to disable some debugging checks when building the libtorrent-rasterbar package.

Fixed version `1.2.11-3` released and currently building on Launchpad.  Please upgrade to it once the builds are done. It may takes 1-2 hours to finish all the builds and be available to download via PPA source. 

https://launchpad.net/~poplite/+archive/ubuntu/qbittorrent-enhanced/+builds?build_text=&build_state=all

**Comment by bakamosokyo** [comment]
> 
> It seems I forgot to disable some debugging checks when building the libtorrent-rasterbar package.
> 
> Fixed version `1.2.11-3` released and currently building on Launchpad. Please upgrade to it once the builds are done. It may takes 1-2 hours to finish all the builds and be available to download via PPA source.
> 

Seems fixed after upgrading.
I will keep this issue open for 3 days.
